sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/IntervalExpressionsSuite.scala

```diff
@@ -535,4 +533,195 @@ class IntervalExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
+      ((Duration.of(-1000, ChronoUnit.MICROS), 0.3f),
+        Array(Duration.ofSeconds(0), Duration.ofSeconds(0),
+          Duration.ofSeconds(0), Duration.ofNanos(-3333000L),
+          Duration.ofSeconds(0), Duration.ofSeconds(0),
+          Duration.ofNanos(-3333000L), Duration.ofSeconds(0),
+          Duration.ofNanos(-3333000L), Duration.ofNanos(-3333000L)),
+        Array(Duration.ofSeconds(0), Duration.ofSeconds(0),
+          Duration.ofSeconds(0), Duration.ofNanos(-300000),
+          Duration.ofSeconds(0), Duration.ofSeconds(0),
+          Duration.ofNanos(-300000), Duration.ofSeconds(0),
+          Duration.ofNanos(-300000), Duration.ofNanos(-300000))),
+      ((Duration.ofDays(9999), 0.0001d),
+        Array(Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L)),
+        Array(Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360))),
+      ((Duration.ofDays(9999), BigDecimal(0.0001)),
+        Array(Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L)),
+        Array(Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360)))
+    ).foreach { case ((duration, num), divideExpected, multiplyExpected) =>
+      DataTypeTestUtils.dayTimeIntervalTypes.zip(divideExpected)
+        .foreach { case (dt, result) =>
+          checkEvaluation(DivideDTInterval(Literal.create(duration, dt), Literal(num)), result)
+        }
+      DataTypeTestUtils.dayTimeIntervalTypes.zip(multiplyExpected)
+        .foreach { case (dt, result) =>
+          checkEvaluation(MultiplyDTInterval(Literal.create(duration, dt), Literal(num)), result)
+        }
+    }
+  }
+
+  test("SPARK-35778: Check multiply/divide of year-month intervals of any fields by numeric 1") {
```

Comment:
Why have two ut for this?

Reply:
I tried to test multiply/divide of day-time intervals of any fields by numeric in two ways. First method is simple, but it is incomplete, because it does not use checkEvaluation. just compare the eval(). Second method is enumerate data to test. I'll delete one way, what about your suggestion? Thank you.